src/modules/mavlink/mavlink_high_latency2.cpp

```diff
@@ -249,12 +274,17 @@ bool MavlinkStreamHighLatency2::write_attitude_sp(mavlink_high_latency2_t *msg)
 bool MavlinkStreamHighLatency2::write_battery_status(mavlink_high_latency2_t *msg)
 {
 	struct battery_status_s battery;
+	bool updated = false;
+
+	for (int i = 0; i < BOARD_NUMBER_BRICKS; i++) {
 
-	const bool updated = _battery_sub->update(&_battery_time, &battery);
+		updated = updated || _battery_sub[i]->update(&_battery_time[i], &battery);
+		_battery_connected[i] = battery.connected;
 
-	if (_battery_time > 0) {
-		if (battery.warning > battery_status_s::BATTERY_WARNING_LOW) {
-			msg->failure_flags |= HL_FAILURE_FLAG_BATTERY;
+		if (_battery_time[i] > 0) {
```

Comment:
Is this not triggered for unset batteries? Just wondering if it's correct.

Reply:
It should be like on Line 537. I have changed it.